#VRML_SIM R2023a utf8

IMPORTABLE EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
IMPORTABLE EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
IMPORTABLE EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
  info [
    "Example use of a LightSensor device."
  ]
  title "Vehicles"
  basicTimeStep 8
}
Viewpoint {
  orientation 0.01038640771865438 -0.9999145567107995 0.00793736811212362 -1.1414853071795894
  position -0.684422 0.0282472 1.54298
}
TexturedBackground {
}
RectangleArena {
  rotation 0 0 1 1.5707963267948966
  floorSize 2 2
}
DEF LAMP Solid {
  translation 0 0.1 0.095
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        roughness 1
        metalness 0
        emissiveColor 1 1 1
      }
      geometry Sphere {
        radius 0.01
      }
    }
    PointLight {
      attenuation 0 0 5
    }
  ]
}
DEF ROBOT Robot {
  translation 0.0185554 -0.367802 -1.77636e-15
  rotation 0 0 1 1.52166
  children [
    Transform {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.04499999999999961 0.025
        rotation 0 -1 0 4.984670000000026
        children [
          DEF WHEEL Transform {
            rotation -1 0 0 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.044999999999999665 0.025
        rotation 0 -1 0 4.587349999999823
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.0385 -0.02 0.063
      rotation 0 0 1 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              rotation -1 0 0 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.02 0.005
                  }
                }
              ]
            }
            Transform {
              rotation -1 0 0 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.02 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation 0.0385 0.02 0.063
      rotation 0 0 1 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation 0.042 0.02 0.07
      rotation 0 0 -1 -0.5
      name "ls0"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    LightSensor {
      translation 0.042 -0.02 0.07
      rotation 0 0 1 -0.5
      name "ls1"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    DEF SMILE Transform {
      translation 0.036 0 0.025
      rotation -0.4472139204829502 -0.7745972408364944 0.44721228048294853 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0 0.0415
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "vehicles"
}
TexturedBackgroundLight {
}
